pkg/ksym/ksym.go

```diff
@@ -204,30 +206,38 @@ func (c *cache) loadKsyms() error {
 			endIndex = len(line)
 		}
 
-		// We care about symbols that are either in the:
-		// - T, t, as they live in the .text (code) section.
-		// - A, means that the symbol value is absolute.
-		//
-		// Add this denylist for symbols that live in the (b)ss section
-		// (d) unitialised data section and (r)ead only data section, in
-		// case there are valid symbols types that we aren't adding.
-		//
-		// See https://linux.die.net/man/1/nm.
-		symbolType := string(line[17:18])
-		if symbolType == "b" || symbolType == "B" || symbolType == "d" ||
-			symbolType == "D" || symbolType == "r" || symbolType == "R" {
-			continue
+		// skip module name
+		name := string(line[19:endIndex])
+		name = strings.TrimFunc(name, unicode.IsSpace)
+		for i, c := range name {
+			if unicode.IsSpace(c) {
+				name = name[:i]
+				break
+			}
 		}
 
-		symbol = string(line[19:endIndex])
-		c.kernelSymbols = append(c.kernelSymbols, ksym{address: address, name: symbol})
```

Comment:
Ideally, this structure should be as small as possible as we keep it in memory, on my machine it's almost 30MBs and according to heap profiles, one of the largest sources of retained memory

Reply:
![截屏2023-02-23 下午7 39 56](https://user-images.githubusercontent.com/52686065/220895823-c94d14a0-e4b3-49ce-b7eb-56230c6a2486.png)

in my machine, this would cost 13 MB of memory, not too much.

Reply:
because we would filter nm type 'a' symbol, so it would contain less memory.

Reply:
What's type "a"? Can't find it here https://man7.org/linux/man-pages/man1/nm.1.html. I wonder if we should filter out the symbols that aren't necessary instead. That would fix the problem, no?